examples/deadline/All-In-AWS-Infrastructure-Basic/python/package/config.py

```diff
@@ -59,6 +59,14 @@ def __init__(self):
         # If false, then we use Amazon DocumentDB to back the render farm.
         self.deploy_mongo_db: bool = False
 
+        # Whether to enable secret management.
+        self.enable_secret_management: bool = True
+
+        # A secret in SecretsManager that stores the admin credentials for secret management.
+        # If not defined and secret management is enabled the secret with admin credentials will be generated.
+        self.secret_management_secret_arn: str =\
+            ''
```

Comment:
Rather than setting this to an empty string, can we make it:
```suggestion
        self.secret_management_secret_arn: Optional[str] = None
```

The check in `service_tier.py` uses `if props.secret_management_secret_arn` and I believe it will always pass if we're using an empty string. Can you double check that we tested all the cases for how this app could be deployed? We should check:
1. SM enabled and no credentials supplied
2. SM enabled and credentials supplied
3. SM disabled
4. SM disabled and credentials supplied (Should fail with message)

Reply:
Did you get a chance to look into this? I think just synthesizing the different cases and looking at the output would be sufficient.

Reply:
yes. I have tested all scenarios except the last one that just warn but continue deployment.